pandas/core/generic.py

```diff
@@ -2967,11 +2967,33 @@ def add_prefix(self, prefix):
 
```

Comment:
`Panel` is deprecated, we may want to use `Series or DataFrame`, or something generic.

I'd try to find uses cases on the internet for this method, and add in the extended summary when this method can be useful, if possible.

Reply:
Thanks! I used Series as it is a Series module in this case.

Reply:
For this and https://github.com/pandas-dev/pandas/pull/20315, these apply to both Series and DataFrame (and Panel, but we don't care about that). So perhaps

    Prefix row labels with a string `prefix`.

I want to avoid `items`, as (to me) I think `dict.items` so key-value pairs. But we're just touching the row labels here.

That summary is strange since we use Prefix as a verb and nound. Maybe "Prepend"  would be better?

Reply:
I see, thank you, my doubt is that on DataFrames add_prefix() adds a prefix on columns names not on rows, for example: `df = pd.DataFrame({'A':[1,2,3,4], 'B':[3,4,5,6]}), df.add_suffix('_item')`. For this motivations I decided to leave on Series. 

Perhaps I could updated the docstring for pandas.DataFrame.add_prefix with the relative example to avoid confusion and let this as it is. 

Let me know what sound better. :)

Reply:
Ahh, good catch! I would say splitting the docstring is more complexity that warranted. How about somehting like

"""
Prefix labels with string `prefix`.

For Series, the row labels are prefixed. For DataFrame, the column labels are prefixed

...
"""

Reply:
Yes, I do agree, thanks!

I didn't know about the issue of splitting docstring, so in this case is better have the same for both pandas.Series.add_prefix and pandas.DataFrame.add_prefix?

Reply:
Yes, the docstring you've been editing in `core/generic.py` is used by both `Series.add_prefix` and `DataFrame.add_prefix`.